libafl/src/mutators/mutations.rs

```diff
@@ -57,6 +54,37 @@ pub fn buffer_set<T: Clone>(data: &mut [T], from: usize, len: usize, val: T) {
     }
 }
 
+/// Generate a range of values where (upon repeated calls) each index is likely to appear in the
+/// provided range as likely as any other value
+///
+/// The solution for this is to specify a window length, then pretend we can start at indices that
+/// would lead to invalid ranges. Then, clamp the values.
+///
+/// This problem corresponds to: <https://oeis.org/A059036>
+#[inline]
+pub fn rand_range<S: HasRand>(state: &mut S, upper: usize, max_len: usize) -> Range<usize> {
+    let len = state.rand_mut().next() as usize % max_len + 1;
+    let offset2 = state.rand_mut().next() as usize % (upper + len);
+    let offset1 = offset2.saturating_sub(len);
+    let offset2 = offset2.clamp(0, upper);
+
+    offset1..offset2
+}
+
+fn rand_range_bucketed<S: HasRand>(state: &mut S, upper: usize, max_len: usize) -> Range<usize> {
+    let max_len = min(
+        match state.rand_mut().next() % 128 {
+            0..=63 => 16,
+            64..=95 => 32,
+            96..=111 => 64,
+            112..=119 => 128,
+            120.. => max_len,
+        },
+        max_len,
+    );
+    rand_range(state, upper, max_len)
+}
+
```

Comment:
This function buckets with 1/2 being <16 bytes, 3/4 being <32 bytes, 7/8 being <64 bytes, 15/16 being <128 bytes, and all inputs being less than the max len provided.

Reply:
Initial results not so positive... Letting it run overnight and we'll see how it goes. Possibly havoc effect of just removing huge sections at once?

Reply:
Okay, while this outperforms the original libafl and the previous mutator fixes on some targets, bucketing is in general worse than the original changes in a 24-hour run. Reverting last commit and then this PR is ready to go.